### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by anuj-kumary**
This PR includes:-

- Sign in Page With Functionality
- Sign up Page With Functionality
- Logout Functionality

Preview:- https://deploy-preview-3--techhub-react.netlify.app/

### CODE REVIEW

**Review by romabulani — APPROVED** [review]
- Code is clean and modular
- Good to merge

File: src/contexts/auth-context.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      });
+    }
+    return () => clearTimeout(id);
+  }, [token]);
+
+  const loginHandler = async (e, setLogin, login) => {
+    e.preventDefault();
+    try {
+      let resp;
+      if (e.target.innerText === 'Sign In as Guest') {
```
Comment (Rohanmond) [inline src/contexts/auth-context.js:28]: could do in a better way! Kinda hardcoding

**Review by Rohanmond — APPROVED** [review]
code looks good, good to see you have used bem conversion in CSS class variable naming

**Review by varnita619 — APPROVED** [review]
Code is clean 
Folder structure is good
Ready to merge

File: src/contexts/auth-context.js (hunk 1 above)
Comment (anuj-kumary) [reply src/contexts/auth-context.js:28]: Sure I'll Explore.

**Pull request merged**